src/vopr.zig

```diff
@@ -0,0 +1,472 @@
+            std.debug.print("rerunning seed {} in Debug mode.\n", .{seed});
+            // Build the simulator binary in Debug mode instead
+            build_simulator(allocator, .Debug);
+            assert(bug == run_simulator(allocator, seed, .Debug, null).?);
+        }
+    }
+    return result;
+}
+
+// Initializes and executes the simulator as a child process.
+// Terminates the VOPR if the simulator fails to run or exits without an exit code.
+fn run_child_process(allocator: mem.Allocator, argv: []const []const u8) u8 {
+    const child_process = std.ChildProcess.init(argv, allocator) catch |err| {
+        fatal("unable to initialize simulator as a child process. Error: {}", .{err});
+    };
+    defer child_process.deinit();
+
+    child_process.stdout = std.io.getStdOut();
+    child_process.stderr = std.io.getStdErr();
+
+    // Using spawn instead of exec because spawn allows output to be streamed instead of buffered.
+    const term = child_process.spawnAndWait() catch |err| {
+        fatal("unable to run the simulator as a child process. Error: {}", .{err});
+    };
+
+    switch (term) {
+        .Exited => |code| {
+            std.debug.print("exit with code: {}\n", .{code});
+            return code;
+        },
+        .Signal => |code| {
+            switch (code) {
+                6 => {
+                    std.debug.print("exit with signal: {}. Indicates a crash bug.\n", .{code});
+                    return @enumToInt(Bug.crash);
+                },
+                else => {
+                    fatal("the simulator exited with an unexpected signal. Term: {}\n", .{term});
+                },
+            }
+        },
+        else => {
+            fatal("the simulator exited without an exit code. Term: {}\n", .{term});
+        },
+    }
+}
+
+fn check_git_status(allocator: mem.Allocator) void {
+    // Running git status to determine whether there is any uncommitted code or local changes.
+    var args = [2][]const u8{ "git", "status" };
+    var exec_result = std.ChildProcess.exec(.{
+        .allocator = allocator,
+        .argv = &args,
+    }) catch |err| {
+        fatal("unable to determine TigerBeetle's git status. Error: {}", .{err});
+    };
+
+    var git_status = exec_result.stdout;
+    if (mem.containsAtLeast(u8, git_status, 1, "nothing to commit, working tree clean") and
+        mem.containsAtLeast(u8, git_status, 1, "Your branch is up to date"))
+    {
+        std.debug.print("All code has been committed and pushed.\n", .{});
+    } else {
+        fatal(
+            "the VOPR cannot run with --send when your branch is ahead or there's uncommited code",
+            .{},
+        );
+    }
+}
+
+// Sends a bug report to the VOPR Hub.
+// The VOPR Hub will attempt to verify the bug and automatically create a GitHub issue.
+fn send_report(allocator: mem.Allocator, address: net.Address, bug: Bug, seed: u64) void {
+    var message: Report = create_report(allocator, bug, seed);
```

Comment:
To match the name better, maybe `send_report` should take a `report` parameter. It would be the caller's responsibility to construct it (with `create_report`). As a bonus, `send_report` wouldn't need an `allocator` anymore.
```suggestion
fn send_report(address: net.Address, report: Report) void {
```